compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -2288,7 +2288,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             }
 
             let field_kind = if is_accessible { "field" } else { "private field" };
-            err.span_label(item_name.span, format!("{field_kind}, not a method"));
+            err.span_label(
+                item_name.span,
+                format!(
+                    "`{item_name}` is a {field_kind} \
+            in the current scope, but is used like a method"
```

Comment:
I think this is pointing to the wrong thing. It doesn't really matter that it's a "field in scope"; it matters that it *is* a method, but that method is not in scope.

I'm not certain there's a good way to detect this case without doing method lookup or recording the right information during method probe.